### PR TITLE
Misc fixes

### LINKED ISSUE #28: ansible installation fail in centos packer boxes with error: Error unpacking rpm package python-crypto-2.6.1-1.el7.centos.x86_64
running the devtest host-group fails while running ansible install task with error: _Error unpacking rpm package python-crypto-2.6.1-1.el7.centos.x86_64_. 

When I try `yum install ansible` on a vm created from the packer box, then I don't see this error.

+++++++
    build-virtualbox:
    build-virtualbox: TASK: [base | install ansible (redhat)] ***************************************
    build-virtualbox: failed: [127.0.0.1] => {"changed": true, "rc": 1, "results": ["Loaded plugins: fastestmirror\nLoading mirror speeds from cached hostfile\n \* base: mirror.beyondhosting.net\n \* epel: fedora-epel.mirror.iweb.com\n \* extras: centos.mb
ni.med.umich.edu\n \* updates: bay.uchicago.edu\nResolving Dependencies\n--> Running transaction check\n---> Package ansible.noarch 0:1.9.4-1.el7 will be installed\n--> Processing Dependency: sshpass for package: ansible-1.9.4-1.el7.noarch\n--> Processin
g Dependency: python-paramiko for package: ansible-1.9.4-1.el7.noarch\n--> Processing Dependency: python-keyczar for package: ansible-1.9.4-1.el7.noarch\n--> Processing Dependency: python-jinja2 for package: ansible-1.9.4-1.el7.noarch\n--> Processing De
pendency: python-httplib2 for package: ansible-1.9.4-1.el7.noarch\n--> Processing Dependency: PyYAML for package: ansible-1.9.4-1.el7.noarch\n--> Running transaction check\n---> Package PyYAML.x86_64 0:3.10-11.el7 will be installed\n--> Processing Depen
dency: libyaml-0.so.2()(64bit) for package: PyYAML-3.10-11.el7.x86_64\n---> Package python-httplib2.noarch 0:0.7.7-3.el7 will be installed\n---> Package python-jinja2.noarch 0:2.7.2-2.el7 will be installed\n--> Processing Dependency: python-babel >= 0.8
 for package: python-jinja2-2.7.2-2.el7.noarch\n--> Processing Dependency: python-markupsafe for package: python-jinja2-2.7.2-2.el7.noarch\n---> Package python-keyczar.noarch 0:0.71c-2.el7 will be installed\n--> Processing Dependency: python-pyasn1 for
package: python-keyczar-0.71c-2.el7.noarch\n--> Processing Dependency: python-crypto for package: python-keyczar-0.71c-2.el7.noarch\n---> Package python-paramiko.noarch 0:1.15.1-1.el7 will be installed\n--> Processing Dependency: python-ecdsa for packag
e: python-paramiko-1.15.1-1.el7.noarch\n---> Package sshpass.x86_64 0:1.05-5.el7 will be installed\n--> Running transaction check\n---> Package libyaml.x86_64 0:0.1.4-11.el7_0 will be installed\n---> Package python-babel.noarch 0:0.9.6-8.el7 will be ins
talled\n---> Package python-crypto.x86_64 0:2.6.1-1.el7.centos will be installed\n---> Package python-ecdsa.noarch 0:0.11-3.el7.centos will be installed\n---> Package python-markupsafe.x86_64 0:0.11-10.el7 will be installed\n---> Package python-pyasn1.n
oarch 0:0.1.6-2.el7 will be installed\n--> Finished Dependency Resolution\n\nDependencies Resolved\n\n================================================================================\n Package                Arch        Version                   Reposit
ory   Size\n================================================================================\nInstalling:\n ansible                noarch      1.9.4-1.el7               epel        1.7 M\nInstalling for dependencies:\n PyYAML                 x86_64
 3.10-11.el7               base        153 k\n libyaml                x86_64      0.1.4-11.el7_0            base         55 k\n python-babel           noarch      0.9.6-8.el7               base        1.4 M\n python-crypto          x86_64      2.6.1-1.e
l7.centos        extras      470 k\n python-ecdsa           noarch      0.11-3.el7.centos         extras       69 k\n python-httplib2        noarch      0.7.7-3.el7               epel         70 k\n python-jinja2          noarch      2.7.2-2.el7
       base        515 k\n python-keyczar         noarch      0.71c-2.el7               epel        218 k\n python-markupsafe      x86_64      0.11-10.el7               base         25 k\n python-paramiko        noarch      1.15.1-1.el7              epe
l        999 k\n python-pyasn1          noarch      0.1.6-2.el7               base         91 k\n sshpass                x86_64      1.05-5.el7                epel         21 k\n\nTransaction Summary\n====================================================
============================\nInstall  1 Package (+12 Dependent packages)\n\nTotal download size: 5.7 M\nInstalled size: 25 M\nDownloading packages:\n--------------------------------------------------------------------------------\nTotal
                              754 kB/s | 5.7 MB  00:07     \nRunning transaction check\nRunning transaction test\nTransaction test succeeded\nRunning transaction\n  Installing : python-crypto-2.6.1-1.el7.centos.x86_64                     1/13 \nerror: u
npacking of archive failed on file /usr/lib64/python2.7/site-packages/pycrypto-2.6.1-py2.7.egg-info: cpio: rename\n  Installing : python-ecdsa-0.11-3.el7.centos.noarch                       2/13 \nerror: python-crypto-2.6.1-1.el7.centos.x86_64: install
failed\n  Installing : python-paramiko-1.15.1-1.el7.noarch                         3/13 \n  Installing : sshpass-1.05-5.el7.x86_64                                   4/13 \n  Installing : python-babel-0.9.6-8.el7.noarch                             5/13 \
n  Installing : python-pyasn1-0.1.6-2.el7.noarch                            6/13 \n  Installing : python-keyczar-0.71c-2.el7.noarch                           7/13 \n  Installing : python-httplib2-0.7.7-3.el7.noarch                          8/13 \n  Inst
alling : python-markupsafe-0.11-10.el7.x86_64                        9/13 \n  Installing : python-jinja2-2.7.2-2.el7.noarch                           10/13 \n  Installing : libyaml-0.1.4-11.el7_0.x86_64                              11/13 \n  Installing
: PyYAML-3.10-11.el7.x86_64                                  12/13 \n  Installing : ansible-1.9.4-1.el7.noarch                                 13/13 \n  Verifying  : python-keyczar-0.71c-2.el7.noarch                           1/13 \n  Verifying  : libya
ml-0.1.4-11.el7_0.x86_64                               2/13 \n  Verifying  : python-jinja2-2.7.2-2.el7.noarch                            3/13 \n  Verifying  : python-markupsafe-0.11-10.el7.x86_64                        4/13 \n  Verifying  : python-httpl
ib2-0.7.7-3.el7.noarch                          5/13 \n  Verifying  : python-pyasn1-0.1.6-2.el7.noarch                            6/13 \n  Verifying  : PyYAML-3.10-11.el7.x86_64                                   7/13 \n  Verifying  : ansible-1.9.4-1.el7
.noarch                                  8/13 \n  Verifying  : python-babel-0.9.6-8.el7.noarch                             9/13 \n  Verifying  : sshpass-1.05-5.el7.x86_64                                  10/13 \n  Verifying  : python-ecdsa-0.11-3.el7.ce
ntos.noarch                      11/13 \n  Verifying  : python-paramiko-1.15.1-1.el7.noarch                        12/13 \n  Verifying  : python-crypto-2.6.1-1.el7.centos.x86_64                    13/13 \n\nInstalled:\n  ansible.noarch 0:1.9.4-1.el7
                                              \n\nDependency Installed:\n  PyYAML.x86_64 0:3.10-11.el7           libyaml.x86_64 0:0.1.4-11.el7_0        \n  python-babel.noarch 0:0.9.6-8.el7     python-ecdsa.noarch 0:0.11-3.el7.centos\n  python-httplib2.
noarch 0:0.7.7-3.el7  python-jinja2.noarch 0:2.7.2-2.el7     \n  python-keyczar.noarch 0:0.71c-2.el7   python-markupsafe.x86_64 0:0.11-10.el7 \n  python-paramiko.noarch 0:1.15.1-1.el7 python-pyasn1.noarch 0:0.1.6-2.el7     \n  sshpass.x86_64 0:1.05-5.el
7          \n\nFailed:\n  python-crypto.x86_64 0:2.6.1-1.el7.centos                                     \n\nComplete!\n"]}
    build-virtualbox: msg: Error unpacking rpm package python-crypto-2.6.1-1.el7.centos.x86_64
    build-virtualbox:

### LINKED ISSUE #32: ERROR! environment must be a dictionary, received env
At the beginning of Ansible playbook run on master node:
TASK [base : ensure custom facts directory exists] *****************************
[DEPRECATION WARNING]: Using bare variables for environment is deprecated. Update your playbooks so that the environment value uses the full variable syntax ('{{foo}}'). This feature will be removed in a future release. Deprecation warnings can be disabled by setting
deprecation_warnings=False in ansible.cfg.
fatal: [node1]: FAILED! => {"failed": true, "msg": "ERROR! environment must be a dictionary, received env (<class 'ansible.parsing.yaml.objects.AnsibleUnicode'>)"}

PLAY RECAP *********************************************************************
node1                      : ok=1    changed=0    unreachable=0    failed=1

—> To correct, you have to change in ./ansible/site.yml all the « environment: env »  to « environment: '{{env}}’ » according to https://github.com/ansible/ansible/issues/11912

Here is a proposed patch. 
[site.txt](https://github.com/contiv/ansible/files/90013/site.txt)

### LINKED ISSUE #34: ceph installation fails due to certificate validation failure
This seems related to proxy, issuing to track it. While commissioning a node in a vm setup on lab host a ceph task is failing

+++++++++
Jan 05 11:37:25 cluster-node1 clusterm[14242]: level=info msg="TASK: [ceph-common | install the ceph stable repository key] ****************** "
Jan 05 11:37:25 cluster-node1 clusterm[14242]: level=info msg="failed: [cluster-node1-0] => {\"failed\": true}"
Jan 05 11:37:25 cluster-node1 clusterm[14242]: level=info msg="msg: SSL Certificate does not belong to download.ceph.com.  Make sure the url has a certificate that belongs to it or use vali
date_certs=False (insecure)"

### PR DESCRIPTION
**Pull request by mapuri**
- workaround for python-crypto install failure (fixes #28 )
  - There seems a know issue (https://bugs.centos.org/view.php?id=9896&nbn=2) with installing python-crypto package that is needed for ansible. And it interferes with ansible's yum installation.

~~\- Also set the ansible version to be installed 1.9~~
~~\- This prevents auto upgrade to 2.0 which is not backward compatible with our playbooks.~~
- turned off certification validation to work behind proxy. (also fixes https://github.com/contiv/cluster/issues/34)
- fixed the syntax for setting environment (fixes #32 )

This fixes some of the issues that I hit while trying to vendor in ansible in build and volplugin. 

/cc @erikh

### CODE REVIEW

**Comment by erikh** [comment]
Ok. I think we should be a little more cautious when turning the `validate_certs` feature off. The reason I think this will become more important is that even though ansible itself cannot handle https proxies, others outside proxies should not have to sacrifice security for it.

That said, we obviously need it. So I'm wondering if we can either:
- detect whether using a proxy and set a variable
- user-supplied variable to turn `validate_certs` off (it really should be on by default)

I think I and our users will be more comfortable with this method. WDYT?

**Comment by erikh** [comment]
Other than the discussion I think this patch LGTM.

**Comment by mapuri** [comment]
@erikh 

> I think I and our users will be more comfortable with this method. WDYT?

yes, even I would be more comfortable with that :). I like the idea of defining a user supplied variable.  It keeps control in user's hands. I will make the change.

**Comment by mapuri** [comment]
@erikh this should be ready for another look.

Please note that I reverted the ansible version change as I realize it will not help us much (our ansible needs to run on hosts with ansible version > 1.9 ) . I instead fixed the `environment` setting that had prompted me make that change in first place.

File: roles/base/tasks/ubuntu_tasks.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@
     - python-pip
 
 - name: add ansible apt repository (debian)
-  apt_repository: repo=ppa:ansible/ansible state=present
+  apt_repository: repo=ppa:ansible/ansible state=present validate_certs=no
```
Comment (erikh) [inline roles/base/tasks/ubuntu_tasks.yml]: you forgot a spot

**Comment by erikh** [comment]
I'm wondering if we should set `validate_certs` (the variable) in group vars instead of in each task? WDYT?

File: roles/base/tasks/ubuntu_tasks.yml (hunk 1 above)
Comment (mapuri) [inline roles/base/tasks/ubuntu_tasks.yml]: oh, oops..fixed it in the updated patch. thanks for catching this!

**Comment by mapuri** [comment]
> I'm wondering if we should set validate_certs (the variable) in group vars instead of in each task? WDYT?

yes, I thought about that. I was in a mixed mind. I felt that ceph related tasks will still need the current way in case we want to upstream this change and avoid having to depend on the group-vars defined there (otherwise they will hit undefined variable error). But thinking more I feel just defining at one place makes it more maintainable and less error prone. Let me make the changes.

**Comment by erikh** [comment]
I think realistically contribution back will get harder and harder, and my initial attempts to provide backports already have failed. The author doesn't seem to want them.

**Comment by mapuri** [comment]
@erikh yes, makes sense. I have made the change to define the variable just once. PTAL when you get a chance.

**Comment by mapuri** [comment]
@erikh I am going to merge this assuming I have incorporated all the comments :). But feel free to post remaining comments, if any.

**Pull request merged**